src/Microsoft.AspNet.BasicMiddleware/OverrideHeaderMiddlewareOptions.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNet.BasicMiddleware
+{
+    public class OverrideHeaderMiddlewareOptions
+    {
+        public OverrideHeaderMiddlewareOptions(bool allOverridesEnabled = true)
+        {
+            XForwardedForEnabled = allOverridesEnabled;
+            XForwardedHostEnabled = allOverridesEnabled;
+            XForwardedProtoEnabled = allOverridesEnabled;
+            XHttpMethodOverrideEnabled = allOverridesEnabled;
+        }
+
+        public bool XForwardedForEnabled { get; set; }
+        public bool XForwardedHostEnabled { get; set; }
+        public bool XForwardedProtoEnabled { get; set; }
+        public bool XHttpMethodOverrideEnabled { get; set; }
```

Comment:
The current implementation fully covers this scenario: http://www.hanselman.com/blog/HTTPPUTOrDELETENotAllowedUseXHTTPMethodOverrideForYourRESTServiceWithASPNETWebAPI.aspx
That seems appropriate for this middleware.